src/source/Common/EcsCredentialProvider.c

```diff
@@ -0,0 +1,290 @@
+/**
+ * Kinesis Video Producer ECS based Credential Provider
+ */
+#define LOG_CLASS "EcsCredentialProvider"
+#include "Include_i.h"
+
+STATUS getHostPort(PCHAR pUrl, PCHAR* ppStart, PCHAR* ppEnd)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    PCHAR pStart = NULL, pEnd = NULL, pCurPtr;
+    UINT32 urlLen;
+    BOOL iterate = TRUE;
+
+    CHK(pUrl != NULL && ppStart != NULL && ppEnd != NULL, STATUS_NULL_ARG);
+
+    // We know for sure url is NULL terminated
+    urlLen = (UINT32) STRLEN(pUrl);
+
+    // Start from the schema delimiter
+    pStart = STRSTR(pUrl, PORT_DELIMITER_STRING);
+    CHK(pStart != NULL, STATUS_INVALID_ARG);
+
+    // Advance the pStart past the delimiter
+    pStart += STRLEN(PORT_DELIMITER_STRING);
```

Comment:
Use ARRAY_SIZE(PORT_DELIMITER_STRING) - 1 to ensure compile time evaluation

Reply:
Sorry for my ignorance, would you mind give me more information about this? I do not understand why we need to change this.

Reply:
STRLEN will be called during the runtime to get the length of the PORT_DELIMITER_STRING which is an extra call and some tooling will complain about not compile-time defining this.
Instead, if you use ARRAY_SIZE macro, it will evaluate to SIZEOF which is sizeof keyword. This will force the compiler to generate the actual value of the length of the string literal at the compile time which is always more desirable.

Reply:
Thanks for your explanation. I changed it to ARRAY_SIZE. :)